perception/autoware_tensorrt_rtmdet/src/trt_batched_nms/batched_nms/trt_batched_nms.cpp

```diff
@@ -0,0 +1,271 @@
+  mPluginAttributes.emplace_back(PluginField("is_normalized", nullptr, PluginFieldType::kINT32, 1));
+  mPluginAttributes.emplace_back(PluginField("clip_boxes", nullptr, PluginFieldType::kINT32, 1));
+  mPluginAttributes.emplace_back(PluginField("return_index", nullptr, PluginFieldType::kINT32, 1));
+
+  mFC.nbFields = mPluginAttributes.size();
+  mFC.fields = mPluginAttributes.data();
+}
+
+const char * TRTBatchedNMSCreator::getPluginName() const TRT_NOEXCEPT
+{
+  return NMS_PLUGIN_NAME;
+}
+
+const char * TRTBatchedNMSCreator::getPluginVersion() const TRT_NOEXCEPT
+{
+  return NMS_PLUGIN_VERSION;
+}
+
+IPluginV2DynamicExt * TRTBatchedNMSCreator::createPlugin(
+  const char * name, const PluginFieldCollection * fc) TRT_NOEXCEPT
+{
+  const PluginField * fields = fc->fields;
+  bool clipBoxes = true;
+  bool returnIndex = false;
+  nvinfer1::plugin::NMSParameters params{};
+
+  for (int i = 0; i < fc->nbFields; ++i) {
+    const char * attrName = fields[i].name;
+    if (!strcmp(attrName, "background_label_id")) {
+      ASSERT(fields[i].type == PluginFieldType::kINT32);
+      params.backgroundLabelId = *(static_cast<const int *>(fields[i].data));
+    } else if (!strcmp(attrName, "num_classes")) {
+      ASSERT(fields[i].type == PluginFieldType::kINT32);
+      params.numClasses = *(static_cast<const int *>(fields[i].data));
+    } else if (!strcmp(attrName, "topk")) {
+      ASSERT(fields[i].type == PluginFieldType::kINT32);
+      params.topK = *(static_cast<const int *>(fields[i].data));
+    } else if (!strcmp(attrName, "keep_topk")) {
+      ASSERT(fields[i].type == PluginFieldType::kINT32);
+      params.keepTopK = *(static_cast<const int *>(fields[i].data));
+    } else if (!strcmp(attrName, "score_threshold")) {
+      ASSERT(fields[i].type == PluginFieldType::kFLOAT32);
+      params.scoreThreshold = *(static_cast<const float *>(fields[i].data));
+    } else if (!strcmp(attrName, "iou_threshold")) {
+      ASSERT(fields[i].type == PluginFieldType::kFLOAT32);
+      params.iouThreshold = *(static_cast<const float *>(fields[i].data));
+    } else if (!strcmp(attrName, "is_normalized")) {
+      params.isNormalized = *(static_cast<const bool *>(fields[i].data));
+    } else if (!strcmp(attrName, "clip_boxes")) {
+      clipBoxes = *(static_cast<const bool *>(fields[i].data));
+    } else if (!strcmp(attrName, "return_index")) {
+      returnIndex = *(static_cast<const bool *>(fields[i].data));
+    }
+  }
+
+  TRTBatchedNMS * plugin = new TRTBatchedNMS(name, params, returnIndex);
+  plugin->setClipParam(clipBoxes);
+  plugin->setPluginNamespace(mNamespace.c_str());
+  return plugin;
+}
+
+IPluginV2DynamicExt * TRTBatchedNMSCreator::deserializePlugin(
+  const char * name, const void * serialData, size_t serialLength) TRT_NOEXCEPT
+{
+  // This object will be deleted when the network is destroyed, which will
+  // call NMS::destroy()
+  TRTBatchedNMS * plugin = new TRTBatchedNMS(name, serialData, serialLength);
```

Comment:
Thank you for your PR!
Please use the smart pointer instead of `new`.
cf. https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#c149-use-unique_ptr-or-shared_ptr-to-avoid-forgetting-to-delete-objects-created-using-new